### PR TITLE
Use CMake 3.2.

### PR DESCRIPTION
**Pull request by pifon2a**

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -31,7 +31,7 @@ google_enable_testing()
 
 find_package(Boost REQUIRED COMPONENTS iostreams)
 find_package(Ceres REQUIRED COMPONENTS SuiteSparse)
-find_package(Eigen3 REQUIRED)
+find_package(Eigen REQUIRED)
```
Comment (gaschler) [inline CMakeLists.txt]: drop this change and rename the FindEigen script

**Comment by pifon2a** [comment]
@wally-the-cartographer merge

**Comment by wally-the-cartographer** [comment]
Merge requested by authorized user pifon2a. Merge queue now has a length of 3.

**Comment by wally-the-cartographer** [comment]
Error: exit status 128

Log:
~~~
=> handle work item: PR 1332 on googlecartographer/cartographer.
 => Switching to branch master.
 $ git reset --hard HEAD [data/cartographer]
HEAD is now at 707f5a5b Merge branch 'master' into migration-old-submap
 $ git clean -fd [data/cartographer]
 $ git checkout master [data/cartographer]
Switched to branch 'master'
Your branch is up to date with 'origin/master'.
 $ git pull [data/cartographer]
Already up to date.
Current branch master is up to date.
 => Checking out pifon2a/cmakecake as local branch pr_1332.
 $ git remote add pifon2a git@github.com:pifon2a/cartographer.git [data/cartographer]
fatal: remote pifon2a already exists.
 $ git fetch pifon2a [data/cartographer]
 $ git branch --track pr_1332 pifon2a/cmakecake [data/cartographer]
fatal: A branch named 'pr_1332' already exists.

~~~

**Pull request merged**